### ISSUE TITLE
CoreWebView2ContextMenuItem submenu crash whole app

### ISSUE THREAD

**Issue by Yannouu** [MicrosoftEdge/WebView2Feedback#1744]
Hello,

When using a `CoreWebView2ContextMenuItem` with Submenu `CoreWebView2ContextMenuItemKind.Submenu`; 
My app crash with _attempted to read or write protected memory_ exception.

**Version**
SDK: 1.0.1010-prerelease
Runtime: 95.0.1015.0
Framework: Win32, WPF
OS: Win10

**Repro Steps**

```
private void CoreWebView2_ContextMenuRequested(object sender, CoreWebView2ContextMenuRequestedEventArgs e){
    CoreWebView2ContextMenuItem c = env.CreateContextMenuItem("submenu", null, CoreWebView2ContextMenuItemKind.Submenu);
    c.CustomItemSelected += contextMenuCustomCommand; //useless?
    CoreWebView2ContextMenuItem subc = env.CreateContextMenuItem("item", null, CoreWebView2ContextMenuItemKind.Command);
    subc.CustomItemSelected += contextMenuCustomCommand;
}

public void contextMenuCustomCommand(object sender, object e)
{
}
```

When I click on the submenu's item the whole app crash with "attempted to read or write protected memory" exception.

Using the same code without submenu works fine.

Only happens on user created submenu; the submenu for traduction works fine.

Regards

[AB#36170328](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/36170328)

**Comment by champnic** [comment]
Thanks for the bug report @Yannouu and for trying out our new API! I've added this bug to our backlog to take a look.

**Comment by yildirimcagri** [comment]
Hi @Yannouu,

I'm looking at this in WPF, but I was not able to repro your crash. My understanding is you are trying to add a custom ContextMenuItem to the WebView2 context menu. Could you share the code where you are actually adding the item to the submenu and the submenu to the WebView2 context menu? A stack of the crash could also be helpful.

Thanks!

**Comment by Yannouu** [comment]
Hello,

I downloaded the WPF sample  (updated 22 sept).
First I changed the BYOWebView2CreationProperties inside App.xaml to use canary 96.0.1025.0 ( contextmenu is not implemented in current evergreen version).
I then updated the webview CreationProperties to use BYOWebView2CreationProperties (I checked in windows task manager).

Then in Main.xaml.cs I add a listerner on ContextMenuRequested inside WebView_CoreWebView2InitializationCompleted.


 ```
void WebView_CoreWebView2InitializationCompleted(object sender, CoreWebView2InitializationCompletedEventArgs e)
{
	if (e.IsSuccess)
	{              /* i removed sample code for readability */
		webView.CoreWebView2.ContextMenuRequested += CoreWebView2_ContextMenuRequested;
		webView.CoreWebView2.FrameCreated += WebView_HandleIFrames;
		return;
	}            
}
```
 



```
 private void CoreWebView2_ContextMenuRequested(object sender, CoreWebView2ContextMenuRequestedEventArgs e)
{
	CoreWebView2ContextMenuItem command = WebViewEnvironment.CreateContextMenuItem("Command", null, CoreWebView2ContextMenuItemKind.Command);
	command.CustomItemSelected += Item_CustomItemSelected;
	e.MenuItems.Add(command);
	CoreWebView2ContextMenuItem menu = WebViewEnvironment.CreateContextMenuItem("MENU", null, CoreWebView2ContextMenuItemKind.Submenu);
	CoreWebView2ContextMenuItem subCommand = WebViewEnvironment.CreateContextMenuItem("subCommand ", null, CoreWebView2ContextMenuItemKind.Command);
	subCommand.CustomItemSelected += Item_CustomItemSelected;
	menu.Children.Add(subCommand);
	e.MenuItems.Add(menu);
}

private void Item_CustomItemSelected(object sender, object e)
{
	return; // do nothing just a test
}

```

It's always crashing  on the **second** use of the sub-command.
The only stack trace VS give me is "External code".
With this exception 
`System.AccessViolationException : 'Attempted to read or write protected memory. This is often an indication that other memory is corrupt.'`


> Also do you see a similar crash in our sample WPF app, in which we are adding a custom submenu ContextMenuItem and an item in it, when you enable the "Custom Context Menu" scenario?

Yes ! 


I tested on 1.0.1010-prerelease and 1.0.1018-prerelease => same result.

Regards;

**Comment by yildirimcagri** [comment]
Thanks @Yannouu, I'm now able to repro the bug and I have a fix in the works. In the meantime, a workaround could be caching and reusing the ContextMenuItem objects you created instead of creating new ones on each ContextMenuRequested event. We actually recommend doing that for performance reasons as well as the fact that you can only create 1000 items per runtime, but I realized the WPF sample doesn't do that. I will also update the sample.

**Comment by Yannouu** [comment]
Hello,

Yes I'll be caching items from now on.

> Thanks again for reporting and let me know if the workaround helped.

To be clear creating the menu only once do not solve the problem, my app still crashs.


This is not the best for some complexe use case, many people using this API will not have a single fixed menu 
=> the menu will depends on what the user is doing, for example my menu will have different items if : 

- the user has selected text
- the user has selected link
- the content is editable 
- etc .....

I guess in **some complexe case** it's just easier to recreate the whole menu on user's request.


Waiting for the fix :-)

Regards,

**Comment by Yannouu** [comment]
Hello,

No more probleme using edge canary Version 96.0.1033.0 and SDK 1.0.1018-prerelease.

Good job ! 
Regards,

**Comment by champnic** [comment]
Thanks for confirming @Yannouu! The fix is available in 96.0.1031.0+ runtimes.
